### ISSUE TITLE
Cannot output to JSON

### ISSUE THREAD

**Issue by scottsilvi** [peterkeating/grunt-csscss#10]
Hey Peter -

Using the following gruntfile, I'm not getting an output file: 

```
'use strict';

module.exports = function(grunt) {

    grunt.initConfig({
        csscss: {
            options: {
                verbose: true,
                outputJson: true
            },
            dist: {
                'output.json' : 'confident.css'
            }
        }
    });

    grunt.loadNpmTasks('grunt-csscss');
    grunt.registerTask('default', ['csscss']);
};
```

If I change the 'output.json' to simply src, I get an output to console, which reads: 

```
Running 'csscss:dist' (csscss) task
confident.css
[truncated json blob ]
Done, without errors
```

Running the above code simply gives: 

```
Running 'csscss:dist' (csscss) task
Done, without errors.
```

Notice the lack of confident.css in the 2nd example...

I'm being asked to clean up a 10k LOC css file I didn't create, and the duplication is rampant, so I would really enjoy working through a JSON file than a console dump that is so long it gets truncated ...

**Comment by peterkeating** [comment]
Hi Scott,

Firstly thanks for creating an issue that easily allowed me to understand the problem your having. With regards to your example `gruntfile.js` not processing any files, apologies, looks like the example in the README doesn't work. By changing the `gruntfile.js` to the one shown below I was able to get the file to be processed.

```
'use strict';

module.exports = function(grunt) {

  grunt.initConfig({
    csscss: {
      dist: {
        options: {
          outputJson: true
        },
        files: {
          'output.json': ['confident.css']
        }
      }
    }
  });

  grunt.loadNpmTasks('grunt-csscss');
  grunt.registerTask('default', ['csscss']);
};
```

If you could give that a go then that would be great! With regards to the truncated JSON blob I'm hoping that solving the outputting to a file will hopefully solve that issue. Otherwise it could be an issue with CSSCSS.

Let us know how you get on.

**Comment by scottsilvi** [comment]
Yeah I was wondering about that ... never seen a grunt plugin that operated in that fashion - figured maybe you just shortcut it or something. 

So now the execution is occurring, but I am getting a "fatal error: unexpected end of input" (the amount of duplication in this 10000 line file is stunning. In a few hours I've chopped 3000 lines out). Is there a file size / selector  limit or some kind of buffer that's being overrun?

**Comment by peterkeating** [comment]
hmm, that's a head scratch! Can you see if the same error occurs if you just run CSSCSS in the command line? Don't suppose the CSS file in question is available so I can recreate what your seeing?

**Comment by scottsilvi** [comment]
Interesting. 

I don't get it if I run csscss -v confident.css. 

I only get this error when outputJson is set to true. If i don't set that, but still pass in a files object, my file is created, just not in JSON format. 

So it has something to do with building a JSON string instead of your default formatting.

**Comment by peterkeating** [comment]
What happens if you run this command, `csscss -v -j confident.css` ?

**Comment by scottsilvi** [comment]
It throws an error: 

Had a problem parsing the css at line: 5517, column: 1
Run with --show-parser-errors for verbose parser errors
Fatal error: Unexepected token

Setting showParserErrors:true reveals: 

Failed to match sequence .... at line 5517 char 1.
-- Don't know what to do with "@media (ma" at line 5517 char 1.

So it looks like this isn't handling media queries.

**Comment by scottsilvi** [comment]
Or I suppose it's possible that in refactoring I removed a close curly at some point. I'll dive in & get back to you when I've confirmed one way or another.

**Comment by peterkeating** [comment]
Yea do keep me posted. Either way you've pointed out some mistakes in the README examples that I need to get on and fix.

**Comment by scottsilvi** [comment]
Alright so I did go ahead and lint the CSS, found a few issues that I've resolved. If I run csscss -v -j confident.css I get a JSON blob. If I run the grunt task, I still get the Fatal error: unexpected end of input. Shall I send you the file?

**Comment by peterkeating** [comment]
Yea that would be really helpful :)

**Comment by peterkeating** [comment]
Hi Scott, I've pushed up [a branch](https://github.com/peterkeating/grunt-csscss/tree/v0.6.2) that should fix the issue. If you could try it out and confirm that would be great.

**Comment by scottsilvi** [comment]
Alright I'll give it a whirl probably in the morning

**Comment by peterkeating** [comment]
You've probably moved on from the CSS file now (hopefully having removed redundancies). Happy for me to close this issue? Will push an update to NPM too.

**Comment by scottsilvi** [comment]
You know I forgot to test this. I'm on vacation now. If you want to close that's fine. I can always respond if it fails.

**Comment by dave-green-uk** [comment]
Hello Peter,

I'm having some trouble that looks to be related to this. I'm setting up my Grunt tasks for the first time as part of a Foundation 5 'boilerplate' for future projects.

The file in question is the compiled Foundation 5 CSS. If I run csscss -v -j app.css from the command line, it spits everything out (a sizeable chunk) with no errors. However, If I run my task using...

```
dist: {
    options: {
        outputJson: true
    },
    files: {
      'output.json': ['css/app.css']
    }
  }
```

...as per your suggestion, I get Fatal error: Unexpected end of input.

Thought I'd share this.

Cheers,
Dave

**Comment by peterkeating** [comment]
I've published v0.6.2 to NPM, so if you could let me know if that fixes the issue or if there are any other issues then that would be great.
